src/python/pants/python/python_setup.py

```diff
@@ -192,22 +151,10 @@ def resolver_allow_prereleases(self):
 
     @property
     def manylinux(self):
-        if self.get_options().resolver_manylinux:
-            manylinux = self.get_options().resolver_manylinux
-            if manylinux.lower() in ("false", "no", "none"):
-                if self.get_options().resolver_use_manylinux:
-                    logger.warning(
-                        "The [{scope}] manylinux option is explicitly set to {manylinux} "
-                        "over-riding the [{scope}] use_manylinux option.".format(
-                            scope=self.options_scope, manylinux=manylinux
-                        )
-                    )
-                return None
-            return manylinux
-        elif self.get_options().resolver_use_manylinux:
-            return self._DEFAULT_MANYLINUX_UPPER_BOUND
-        else:
+        manylinux = self.get_options().resolver_manylinux
+        if manylinux is None or manylinux.lower() in ("false", "no", "none"):
```

Comment:
We could potentially consider returning an `Enum` here:
```python
class ManylinuxOption(Enum):
  false = 'false'
  no = 'no'
  none = 'none'
```
Which could then be used as the type of the option.

This is a note for the future and **not** at all a blocking comment.

Reply:
The option also accepts things like manylinux1, manylinux2010 and manylinux 2014 today. Could also add those to an option, but then we're tied to otherwise pass-through string values to Pex which are in turn pass-through string values to Pip which are in turn pass-through string values to packaging.

Reply:
Thank you! That makes sense.